components/api/src/main/java/com/hotels/styx/api/plugins/spi/PluginMeterRegistry.java

```diff
@@ -0,0 +1,157 @@
+/*
+  Copyright (C) 2013-2020 Expedia Inc.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+ */
+package com.hotels.styx.api.plugins.spi;
+
+import io.micrometer.core.annotation.Incubating;
+import io.micrometer.core.instrument.Counter;
+import io.micrometer.core.instrument.DistributionSummary;
+import io.micrometer.core.instrument.Meter;
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.Tag;
+import io.micrometer.core.instrument.Tags;
+import io.micrometer.core.instrument.Timer;
+import io.micrometer.core.instrument.search.RequiredSearch;
+import io.micrometer.core.instrument.search.Search;
+import io.micrometer.core.lang.Nullable;
+
+import java.util.Collection;
+import java.util.Map;
+import java.util.function.Consumer;
+import java.util.function.ToDoubleFunction;
+
+public class PluginMeterRegistry {
+    public static final String DEFAULT_TAG_KEY = "plugin";
+    private final MeterRegistry meterRegistry;
+    private final Tag defaultTag;
```

Comment:
I'd suggest calling this (and related constants/methods) `pluginTag`, it's more meaningful.

Reply:
I originally started with the name `pluginTag`, but then changed it to `defaultTag` to give the notion of how this tag is **mandatory** and **injected** by default without any intervention needed. What do you think of it?

Reply:
On second thought, I will rename it to `defaultPluginTag`, what do you say?

Reply:
On second thought, I will rename it to defaultPluginTag, how does that sound?